### PR TITLE
CRM-21851 Current employer dissapears when disabling expired relationships

### PR DESCRIPTION
**Pull request by anaskoara**
To fix this we have to check if there is no unexpired relation ship before remove current contact employer.

### CODE REVIEW

File: CRM/Contact/BAO/Relationship.php [hunk 1]
```diff
@@ -675,7 +675,13 @@ public static function clearCurrentEmployer($id, $action) {
         $employerRelTypeId = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Employee of', 'id', 'name_a_b');
 
         if ($relationship->relationship_type_id == $employerRelTypeId && $relationship->contact_id_b == $sharedContact->employer_id) {
-          CRM_Contact_BAO_Contact_Utils::clearCurrentEmployer($relationship->contact_id_a);
+          $currentEmployerRlationship = new CRM_Contact_DAO_Relationship();
```
Comment (seamuslee001) [inline CRM/Contact/BAO/Relationship.php]: @anaskoara the indenting in these lines looks wrong

File: CRM/Contact/BAO/Relationship.php [hunk 2]
```diff
@@ -2271,6 +2277,15 @@ public static function isCurrentEmployerNeedingToBeCleared($params, $relationshi
     if ($existingTypeName !== 'Employer of') {
       return FALSE;
     }
+	//check if there are another relationship which is not expired 
+	  $relationship = new CRM_Contact_DAO_Relationship();
+      $relationship->contact_id_a = $params['contactID'];
+      $relationship->relationship_type_id = $params['relationship_type_id'];
+	  $relationship->end_date = NULL;
```
Comment (seamuslee001) [inline CRM/Contact/BAO/Relationship.php]: As with these lines

**Comment by eileenmcnaughton** [comment]
@jitendrapurohit are you able to look at this - you dealt with related code back in May & may be well placed to assess

**Comment by jitendrapurohit** [comment]
Don't think there is an issue here.  @anaskoara Can you pls elaborate or mention the steps to reproduce the problem? 

AFAIK, the current employer should not be shown if a contact has an expired relationship with an organization. So not sure what you mean by "disabling expired relationships". Expired relationships are already listed under "Inactive" section of the relationship tab and disabling it from the "Edit" form does not change the value of the current employer stored for the contact. In both the cases, the value should be empty. Isn't the same happening for you?

**Comment by francescbassas** [comment]
Steps to reproduce the problem described here https://lab.civicrm.org/dev/core/issues/470. I'm not sure if is exactly the same issue that @anaskoara tries to fix.

>1. Create a current relationship on individual A as employee of organization B and mark as current employer
>2. Create a past relationship on individual A as employee of organization B without marking as current employer
>
>At this point on summary tab of individual A appears organization B as his Employer.
>
>3. Run Disable expired relationships cron job
>
>At this point on summary tab there is no organization at Employer field although there is one active employee relationship and past relationship was disabled.

**Comment by jitendrapurohit** [comment]
Thanks for the gitlab link @francescbassas. I can replicate it now and will add the review notes after testing it a bit 👍

File: CRM/Contact/BAO/Relationship.php [hunk 3]
```diff
@@ -675,7 +675,13 @@ public static function clearCurrentEmployer($id, $action) {
...
 
         if ($relationship->relationship_type_id == $employerRelTypeId && $relationship->contact_id_b == $sharedContact->employer_id) {
-          CRM_Contact_BAO_Contact_Utils::clearCurrentEmployer($relationship->contact_id_a);
+          $currentEmployerRlationship = new CRM_Contact_DAO_Relationship();
+          $currentEmployerRlationship->contact_id_a = $relationship->contact_id_a;
+          $currentEmployerRlationship->end_date=NULL;
+          $currentEmployerRlationship->is_active=1;
+          if(!$currentEmployerRlationship->find(TRUE)){
+	     CRM_Contact_BAO_Contact_Utils::clearCurrentEmployer($relationship->contact_id_a);
+          }
```
Comment (jitendrapurohit) [inline CRM/Contact/BAO/Relationship.php]: I think the above checking can be done by an existing function in this file. @anaskoara Can you check if you can achieve the same outcome by using the if condition as shown below?

    $existingCurrentEmployerRelationship = self::getRelationship($relationship->contact_id_a, self::CURRENT, 0, 0, 0, NULL, NULL, FALSE, array('relationship_type_id' => $employerRelTypeId));

    if (!empty($existingCurrentEmployerRelationship)) {
      //there is an active relationship, so avoid clearing the employer
    }

**Comment by monishdeb** [comment]
Hey guys, I am closing this PR in favor of https://github.com/civicrm/civicrm-core/pull/13334 after cherry-picking the commits from this PR. Made some additional fixes and added UT. 
Please have a look.

**Pull request closed without merging**